gabbi/json_parser.py

```diff
@@ -0,0 +1,51 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import jsonpath_rw
+
+
+class Len(jsonpath_rw.JSONPath):
+    """The JSONPath referring to the len of the current object.
+
+    Concrete syntax is '`len`'.
+    """
+
+    def find(self, datum):
+        datum = jsonpath_rw.DatumInContext.wrap(datum)
+        return [jsonpath_rw.DatumInContext(len(datum.value),
```

Comment:
I don't have the brains to parse the code in detail right now or think deeply about the conceptual implications. However your suggestion WRT type agnosticism makes sense. In fact, I can imagine users simply assume that it works for any iterable, so not supporting that would likely end up being surprising.